src/backend/crossterm.rs

```diff
@@ -4,28 +4,140 @@
+    },
+    terminal,
+    terminal::{Clear, ClearType},
+    Output,
 };
 
-use crate::event::{Event, Key, MouseButton, MouseEvent};
-use std::cell::{Cell, RefCell};
-use std::io::{self, BufWriter, Stdout, Write};
+use crate::{
+    backend,
+    event::{Event, Key, MouseButton, MouseEvent},
+    theme,
+    vec::Vec2,
+};
+use crossterm::event::KeyCode;
 
 /// Backend using crossterm
 pub struct Backend {
     current_style: Cell<theme::ColorPair>,
     last_button: Option<MouseButton>,
     // reader to read user input async.
-    async_reader: AsyncReader,
     _alternate_screen: AlternateScreen,
     stdout: RefCell<BufWriter<Stdout>>,
-    terminal: Terminal,
+}
+
+impl From<CMouseButton> for MouseButton {
+    fn from(button: CMouseButton) -> Self {
+        match button {
+            CMouseButton::Left => MouseButton::Left,
+            CMouseButton::Right => MouseButton::Right,
+            CMouseButton::Middle => MouseButton::Middle,
+        }
+    }
+}
+
+impl From<KeyCode> for Key {
+    fn from(code: KeyCode) -> Self {
+        match code {
+            KeyCode::Esc => Key::Esc,
+            KeyCode::Backspace => Key::Backspace,
+            KeyCode::Left => Key::Left,
+            KeyCode::Right => Key::Right,
+            KeyCode::Up => Key::Up,
+            KeyCode::Down => Key::Down,
+            KeyCode::Home => Key::Home,
+            KeyCode::End => Key::End,
+            KeyCode::PageUp => Key::PageUp,
+            KeyCode::PageDown => Key::PageDown,
+            KeyCode::Delete => Key::Del,
+            KeyCode::Insert => Key::Ins,
+            KeyCode::Enter => Key::Enter,
+            KeyCode::Tab => Key::Tab,
+            KeyCode::F(n) => Key::from_f(n),
+            KeyCode::BackTab => Key::Tab, /* not supported */
+            KeyCode::Char(_) => Key::Tab, /* is handled at `Event` level, use tab as default */
+            KeyCode::Null => Key::Tab, /* is handled at `Event` level, use tab as default */
+        }
+    }
+}
+
+impl From<CKeyEvent> for Event {
+    fn from(event: CKeyEvent) -> Self {
+        const CTRL_ALT: KeyModifiers = KeyModifiers::from_bits_truncate(
+            KeyModifiers::CONTROL.bits() | KeyModifiers::ALT.bits(),
+        );
+        const CTRL_SHIFT: KeyModifiers = KeyModifiers::from_bits_truncate(
+            KeyModifiers::CONTROL.bits() | KeyModifiers::SHIFT.bits(),
+        );
+        const ALT_SHIFT: KeyModifiers = KeyModifiers::from_bits_truncate(
+            KeyModifiers::ALT.bits() | KeyModifiers::SHIFT.bits(),
+        );
+
+        match event {
+            CKeyEvent {
+                modifiers: KeyModifiers::CONTROL,
+                code: KeyCode::Char(c),
+            } => Event::CtrlChar(c),
+            CKeyEvent {
+                modifiers: KeyModifiers::ALT,
+                code: KeyCode::Char(c),
+            } => Event::AltChar(c),
+            CKeyEvent {
+                modifiers: _,
+                code: KeyCode::Char('c'),
+            } => Event::Exit,
```

Comment:
Maybe we want exit for `Ctrl-C`, not for just `'c'`?

Reply:
So a capital C?

Reply:
I mean, right now, doesn't this mean that pressing the letter <kbd>c</kbd> will close the program? Is that desired?

Reply:
Correct, I looked at [this](https://github.com/gyscos/cursive/blob/master/src/backend/termion.rs#L123) it also has Ctrl + c. I will do an update